src/main/java/io/github/thebusybiscuit/slimefun4/core/services/profiler/PerformanceSummary.java

```diff
@@ -0,0 +1,157 @@
+package io.github.thebusybiscuit.slimefun4.core.services.profiler;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+
+import io.github.thebusybiscuit.cscorelib2.chat.ChatColors;
+import io.github.thebusybiscuit.slimefun4.utils.ChatUtils;
+import io.github.thebusybiscuit.slimefun4.utils.NumberUtils;
+import net.md_5.bungee.api.ChatColor;
+import net.md_5.bungee.api.chat.HoverEvent;
+import net.md_5.bungee.api.chat.TextComponent;
+
+class PerformanceSummary {
+
+    // The threshold at which a Block or Chunk is significant enough to appear in /sf timings
+    private static final int VISIBILITY_THRESHOLD = 275_000;
+
+    // A minecraft server tick is 50ms and Slimefun ticks are stretched across
+    // two ticks (sync and async blocks), so we use 100ms as a reference here
+    static final int MAX_TICK_DURATION = 100;
+
+    private final SlimefunProfiler profiler;
+    private final PerformanceRating rating;
+    private final long totalElapsedTime;
+    private final int totalTickedBlocks;
+
+    private final Map<String, Long> chunks;
+    private final Map<String, Long> items;
+
+    PerformanceSummary(SlimefunProfiler profiler, long totalElapsedTime, int totalTickedBlocks) {
+        this.profiler = profiler;
+        this.rating = profiler.getPerformance();
+        this.totalElapsedTime = totalElapsedTime;
+        this.totalTickedBlocks = totalTickedBlocks;
+
+        chunks = profiler.getByChunk();
+        items = profiler.getByItem();
+    }
+
+    public void send(CommandSender sender) {
+        sender.sendMessage("");
+        sender.sendMessage(ChatColor.GREEN + "===== Slimefun Lag Profiler =====");
+        sender.sendMessage(ChatColors.color("&6Total: &e" + NumberUtils.getAsMillis(totalElapsedTime)));
+        sender.sendMessage(ChatColors.color("&6Performance: " + getPerformanceRating()));
+        sender.sendMessage(ChatColors.color("&6Active Chunks: &e" + chunks.size()));
+        sender.sendMessage(ChatColors.color("&6Active Blocks: &e" + totalTickedBlocks));
+        sender.sendMessage("");
```

Comment:
\n is your friend

Reply:
Yeah not gonna convert this one really. Having a StringBuilder for this would just be extremely ugly here.